pycbc/detector.py

```diff
@@ -82,21 +83,29 @@ def __init__(self, detector_name, reference_time=1126259462.0):
 
         """
         self.name = str(detector_name)
-        self.frDetector =  lalsimulation.DetectorPrefixToLALDetector(self.name)
+        self.frDetector = lalsimulation.DetectorPrefixToLALDetector(self.name)
         self.response = self.frDetector.response
         self.location = self.frDetector.location
         self.latitude = self.frDetector.frDetector.vertexLatitudeRadians
         self.longitude = self.frDetector.frDetector.vertexLongitudeRadians
 
         self.reference_time = reference_time
-        if reference_time is not None:
+        self.sday = None
+        self.gmst_reference = None
+
+    def set_gmst_reference(self):
+        if self.reference_time is not None:
             self.sday = float(sday.si.scale)
-            self.gmst_reference = gmst_accurate(reference_time)
+            self.gmst_reference = gmst_accurate(self.reference_time)
+        else:
+            raise RuntimeError("Can't get accurate sidereal time without GPS "
+                               "reference time!")
 
     def gmst_estimate(self, gps_time):
         if self.reference_time is None:
             return gmst_accurate(gps_time)
 
+        self.set_gmst_reference()
```

Comment:
This should only be called if the gmst_reference isn't already set. We don't want to call gmst_accurate more than once if the reference time is provided.

Reply:
I think you may just need another if statment here that checks if the gmst_reference is already set or not, so that it doesn't reset it each time.